### DISCUSSION TITLE
-std=c++20

### DISCUSSION THREAD

**Discussion by evilfrog13** [aristocratos/btop#182] [Q&A] [answered]
hi im new to all this im on ubuntu 20.04 LTS and wen i try the command MAKE i get 
g++: error: unrecognized command line option '-std=c++20'; did you mean 'std=c++2a'?
thx

**Comment by userdocs** [comment, marked as answer]
**edit:**  you probably need to use gcc 10 anyway as that is the minimum required to build btop

I believe you have to use either specify this `-std=c++2a` with gcc 9 or earlier (the default in ubuntu focal) or use [gcc 10](https://packages.ubuntu.com/focal/gcc-10)

From google: 

C++20 features are available since GCC 8.

To enable C++20 support, add the command-line parameter

`-std=c++20`

For G++ 9 and earlier use

`-std=c++2a`

Or, to enable GNU extensions in addition to C++20 features, add

`-std=gnu++20`

**Reply by evilfrog13** [reply]
im really new to this gcc 10 is installed wen i got that error and i don't know where to use those command you are mentioning the only solution that worked for me was doing this
sudo add-apt-repository ppa:ubuntu-toolchain-r/test
and updating everything after

**Reply by nsh07** [reply]
@evilfrog13 if you have the `g++-10` package installed from apt, you'll need to run `make CXX=g++-10` instead of the usual `make` because the default C++ compiler on 20.04 is g++-9, which doesn't support the `-std=c++20` option.
Adding PPAs is unnecessary since GCC 10 is in the Ubuntu repos

**Reply by anmac1789** [reply]
How to install gcc10 with Ubuntu 20.04 or how do we type the command `-std=c++20` OR `-std=c++2a`

**Comment by evilfrog13** [comment]
i was off this project for a while and still dont know what i am doing will make some change when i canOn Oct 28, 2025, at 2:36 PM, anmac1789 ***@***.***> wrote:﻿
How to install gcc10 with Ubuntu 20.04 or how do we type the command -std=c++20 OR -std=c++2a

**Reply by anmac1789** [reply]
im sorry I think you meant to quote my last reply but it got mixed in with your original message...
